src/irmin-pack/layout.ml

```diff
@@ -40,8 +40,15 @@ module V3 = struct
   let sorted ~generation =
     toplevel ("store." ^ string_of_int generation ^ ".sorted")
 
-  let mapping ~generation =
-    toplevel ("store." ^ string_of_int generation ^ ".mapping")
+  let mapping =
+    (* if we use mmap for the mapping file, then the mapping file is not portable between
+       architectures with different endianness; to avoid the possibility that a
+       little-endian file is used on a big-endian arch (or vice versa) we use a filename
+       that is not portable between archs; since the common case is little endian, we make
+       this the "default" empty suffix, and for big endian we add ".be" *)
+    let empty_or_be = match Sys.big_endian with false -> "" | true -> ".be" in
```

Comment:
I'm a bit worried about endianness here (sorry, I forgot that mmap required this). 

For instance, s390x uses big endian, and this is something we were asked to fix in the past https://github.com/mirage/irmin/pull/1694.

Another issue I see, is that if at some point we change the snapshot format, to be the upper layer (that is just a zip of the store after a gc) that means we need to provide some guarantees of portability of the mapping file too. So again, we give up support for big-endian machines, maybe its fine?

Reply:
Hi yes we talked about this at various points. My proposal would be: use mmap and a non-portable file format, but convert if we find ourselves on a different architecture. If, for example, we run on big-endian and find the mapping file is missing, we look for the little-endian version and convert it, then carry on. This presumably only happens at startup, so is effectively invisible to the user, and allows portable context directories.

Reply:
Another solution would be to keep only saving the LE format, and then perform an on-the-fly conversion using `Bytes. get_int64_le` on BE platforms. 

This way the code stays portable, it keeps the perfs on LE platforms and induces a very minor overhead on BE platforms